### PR TITLE
Initial commit of CVEs 2010-4488 & 2014-7937

### PR DESCRIPTION
**Pull request by rl5849**
Initial committ of CVEs 2010-4488 & 2014-7937

### CODE REVIEW

File: cves/CVE-2014-7937.yml [hunk 1]
```diff
@@ -93,20 +98,24 @@ discovered:
...
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: |
+    ClusterFuzz, a google produced fuzzer, verified and reported on the
+    bug.
+  date: 2014-09-30
+  automated: true
+  google: yes
```
[the remark this answers is not in the document]
Comment (rl5849) [reply cves/CVE-2014-7937.yml]: Thanks, I don't know how the CI and I missed that.

File: cves/CVE-2014-7937.yml [hunk 2]
```diff
@@ -127,12 +138,15 @@ major_events:
 
     The event doesn't need to be directly related to this vulnerability, rather,
     we want to capture what the development team was dealing with at the time.
-  answer: 
+  answer: |
+    The past three commits to this one were all affecting this file. There
```
[the remark this answers is not in the document]
Comment (rl5849) [reply cves/CVE-2014-7937.yml]: I was referencing the fix, sorry I didn't make that clear, I have reworded this, thank you.

File: cves/CVE-2010-4488.yml [hunk 3]
```diff
@@ -32,7 +32,11 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
```
Comment (jtboundy) [inline cves/CVE-2010-4488.yml]: I think you did a really good job at explaining what the vulnerability was. I did notice a spelling error for the word 'ultimately' on the last line however.

File: cves/CVE-2010-4488.yml [hunk 4]
```diff
@@ -54,16 +58,16 @@ fixes_vcc_instructions: |
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
 - :commit: 7a67a81536b664eb4145731ec1e97bf2f7990e30
-  :note: ''
-vccs: []
+  :note: The VCC was the second commit since the initial commit. This was an oversight in the initial development
```
Comment (jtboundy) [inline cves/CVE-2010-4488.yml]: Is there any more information on the oversight that caused the vcc?

File: cves/CVE-2010-4488.yml [hunk 5]
```diff
@@ -94,20 +98,28 @@ discovered:
...
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: |
+    The bug was discovered by someone working at denyall.com
+    who stubbled upon the bug and reported it. They were using a proxy and were
```
Comment (jtboundy) [inline cves/CVE-2010-4488.yml]: Good answer. Just noticed spelling error for 'stumbling'.

File: cves/CVE-2014-7937.yml [hunk 6]
```diff
@@ -32,7 +32,12 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  Multiple off-by-one errors in libavcodec/vorbisdec.c resulting for-loop over/under-execution.
```
Comment (jtboundy) [inline cves/CVE-2014-7937.yml]: Maybe provide more detail on what libavcodec/vorbisdec.c does for the system overall.

File: cves/CVE-2014-7937.yml [hunk 7]
```diff
@@ -149,37 +163,39 @@ lessons:
...
   native_wrappers:
-    applies: 
+    applies: false
     note: 
   distrust_input:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      Crafted Vorbis I data could be put into the code causing 'other impacts', data should scubbed/escaped.
```
Comment (jtboundy) [inline cves/CVE-2014-7937.yml]: Spelling error on 'scrubbed'.

File: cves/CVE-2014-7937.yml [hunk 8]
```diff
@@ -193,4 +209,9 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    This error was cause by a simple coding mistake, an off-by-one error. Theses mistakes are bound to happen,
+    the only thing that can prevents error like this are extensive tests that include boundary and edge
+    cases. The mitigation is very simple and does look proper. An off-by-one is quite easy to identify
+    and correct. This was fixed according to the CWE, as the fix is making sure loops stay within range, when
+    working in languages like C.
```
[the remark this answers is not in the document]
Comment (jtboundy) [reply cves/CVE-2014-7937.yml]: Agree that this explanation is a good description of how the mistake happened. Will also give this 1 upvote.

File: cves/CVE-2010-4488.yml [hunk 9]
```diff
@@ -194,4 +210,12 @@ mistakes:
...
-  answer: 
+  answer: |
+    This could be a coding or design error, there was obviously no
+    system designed for handling this case. However, when coding this, the developer
+    did not see the possibility of failure and all it took to fix this was an if condition
+    to test whether the connection behind a proxy succeeded. The fix does adhere to the
+    CWE mitigations and the fix does appears to be correct. However, in more recent
+    versions of the file, the original fix is not present. Handling this scenario
+    is still present, just in a differnt form, this makes it seem that the original fix
+    was not as rock-solid as it should have been.
```
Comment (jtboundy) [inline cves/CVE-2010-4488.yml]: I think that this is a good description of how the mistake happened. I thought this was pretty interesting and will give it 3 upvotes.

File: cves/CVE-2010-4488.yml (hunk 3 above)
Comment (rl5849) [reply cves/CVE-2010-4488.yml]: Fixed, thank you.

File: cves/CVE-2010-4488.yml (hunk 4 above)
Comment (rl5849) [reply cves/CVE-2010-4488.yml]: I have updated the notes, this vulnerabilty is a coding and design error, the team just did not account for this case so early in the project.

File: cves/CVE-2010-4488.yml (hunk 5 above)
Comment (rl5849) [reply cves/CVE-2010-4488.yml]: Fixed, thank you.

File: cves/CVE-2014-7937.yml (hunk 6 above)
Comment (rl5849) [reply cves/CVE-2014-7937.yml]: Good idea, what I found was quite interesting, thanks!

File: cves/CVE-2014-7937.yml (hunk 7 above)
Comment (rl5849) [reply cves/CVE-2014-7937.yml]: Fixed, thank you.

**Pull request merged**